proposals/0327-actor-initializers.md

```diff
@@ -211,343 +211,593 @@ struct S {
 
 ```swift
-actor A {
-  var friends: [A]
+actor Bob {
+  var x: Int
+  var y: Int = 2
+  func f() {}
+  init(_ cond: Bool) async {
+    if cond {
+      self.x = 1 // initializing store
+    }
+    self.x = 2 // initializing store
 
-  init(withFriends fs: [A]) {
-    friends = fs
-    self.notifyAll()  // ❌ disallowed by escaping-use restriction.
+    f() // this is ok, since we're on the executor here.
   }
+}
+```
 
-  @MainActor
-  init() {
-    friends = ...
-    self.notifyAll()  // ❌ disallowed by escaping-use restriction.
+The problem with trying to explicitly mark the suspension points in `Bob.init` is that they are not easy for programmers to track, nor are they consistent enough to stay the same under simple refactorings. Adding or removing a default value for a stored property, or changing the number of stored properties, can greatly influence where the hops may occur. Consider this slightly modified example from before:
+
+```swift
+actor EvolvedBob {
+  var x: Int
+  var y: Int
+  func f() {}
+  init(_ cond: Bool) async {
+    if cond {
+      self.x = 1
+    }
+    self.x = 2 
+    self.y = 2 // initializing store
+
+    f() // this is ok, since we're on the executor here.
   }
+}
+```
 
-  func verify() { ... }
-  func notifyAll() { ... }
+Relative to `Bob`, the only change made to `EvolvedBob` is that its default value for `y` was converted into an unconditional store in the body of the initializer. From an observational point of view, `Bob.init` and `EvolvedBob.init` are identical. But from an implementation perspective, the suspension points for performing an executor hop differ dramatically. If those points required some sort of annotation in Swift, such as with `await`, then the reason why those suspension points moved is hard to explain to programmers.
+
+In summary, we propose to _implicitly_ perform suspensions to hop to the actors executor once `self` is initialized, instead of having programmers mark those points explicitly, for the following reasons:
+1. The finding and continually updating the suspension points is annoying for programmers.
+2. The reason _why_ some simple stores to a property can trigger a suspension is an implementation detail that is hard to explain to programmers.
+3. The benefits of marking these suspensions is very low. The reference to `self` is known to be unique by the time the suspension  will happen, so it is impossible to create an [actor reentrancy](https://github.com/apple/swift-evolution/blob/main/proposals/0306-actors.md#actor-reentrancy) situation.
```

Comment:
I agree though there are situations where we can cause reentrancy during the deinit -- if we escaped self. I don't think this counters your point here though. No need to add info about escaping here... I think you're right that the simple things don't violate anything

Reply:
The `deinit` isn't gaining any implicit hops in its body. For an `init`, you can't escape `self` before the hop happens. I edited some of the wording earlier in this section to highlight that.